pkg/handlers/v1/fetchbyip.go

```diff
@@ -0,0 +1,78 @@
+package v1
+
+import (
+	"context"
+	"net"
+
+	"github.com/asecurityteam/ipam-facade/pkg/domain"
+	"github.com/asecurityteam/ipam-facade/pkg/logs"
+)
+
+// IPAddressQuery contains an IP address on which to search for physical assets.
+type IPAddressQuery struct {
+	IPAddress string `json:"ipAddress"`
+}
+
+// PhysicalAssetDetails provides the response structure for PhysicalAsset records returned from storage.
+type PhysicalAssetDetails struct {
+	IP            string `json:"ip"`
+	ResourceOwner string `json:"resourceOwner"`
+	BusinessUnit  string `json:"businessUnit"`
+	Tags          tags   `json:"tags"`
+}
+
+// tags is the key-value pair structure that provides less important information than the
+// root keys of the PhysicalAssetDetails response.
+type tags struct {
+	SubnetID string `json:"subnetID"`
+	Network  string `json:"network"`
+	DeviceID string `json:"deviceID"`
+	Location string `json:"location"`
+}
+
+// FetchByIPAddressHandler uses its PhysicalAssetFetcher implementation to serve fetch requests for
+// physical assets by IP address.
+type FetchByIPAddressHandler struct {
+	PhysicalAssetFetcher domain.PhysicalAssetFetcher
+	LogFn                domain.LogFn
+}
+
+// Handle processes an incoming IPAddressQuery and returns a PhysicalAssetDetails response or an error.
+func (h *FetchByIPAddressHandler) Handle(ctx context.Context, query IPAddressQuery) (PhysicalAssetDetails, error) {
+	logger := h.LogFn(ctx)
+
+	if ip := net.ParseIP(query.IPAddress); ip == nil {
+		err := domain.InvalidInput{IP: query.IPAddress}
+		logger.Error(logs.InvalidInput{Reason: err.Error()})
```

Comment:
suggestion: Since this is driven from client input, I think Info may help keep log noise to a minimum.

Reply:
From my perspective, a 4XX should **never** happen. That either means someone we don't expect is calling this service, or one of our client services is misconfigured, and so it seems like we should maximize their visibility. I would very much like to know your view, though, so let's discuss!

Reply:
> That either means someone we don't expect is calling this service

How would invalid input indicate this? Would the gateway authentication/access logs be a better indicator of this?

> or one of our client services is misconfigured

It sounds like this would be a fatal error in that service, and not necessarily this service

The way that I am thinking about this is - is this outcome the result some sort of actionable error in this service? In this case, it seems like this outcome (400), would indicate an actionable error in _another_ service. In that case, I would expect an error log to be present in the calling service.

Reply:
Ah, ok. I agree on both points, and will make the change. Thanks!!